Test that Web Dispatch logs source IP address and path

Dzol opened the pull request:
This is general enough to test Web Dispatch with every application that does logging as it's done here. Indeed, the Management plugin _does_ log like this, so this closes https://github.com/rabbitmq/rabbitmq-server/issues/483.

Unlike the output given for issue #483 at https://groups.google.com/d/msg/rabbitmq-users/CBUYqqBwlg0/nsQwQD8HBQAJ the logging tested here doesn't log a **reason**, but it does log a **source** IP and **path**. Logging a **reason** wouldn't be much work.

Feedback on the directory the log is written out to would be appreciated.

Commit:
Test that Web Dispatch logs source IP address and path

This is general enough to test Web Dispatch with every application
that does logging as it's done here. Indeed, the Management plugin
does log like this, so this closes #483.

michaelklishin commented:
@Dzol HTTP API log location is [configurable](http://www.rabbitmq.com/management.html).

Dzol commented:
> HTTP API log location is configurable.

The test first configures things so that a log is written to a file in `/tmp/rabbit/test` which is read later. So it should test that logging (with a source IP address) is done for anyone who uses Web Dispatch in such a way. Shall I write a test just for the Management plugin?

Commit:
Test that a given status is logged

michaelklishin commented:
OK, I understand the question better now. Take a look at what test directory is used by `rabbitmq-test`. Off top of my head, it is `$TMPDIR/rabbit-test/log/*` but those differ in multi-node tests.

michaelklishin commented:
@Dzol another thing to clarify: this only submits a test. Does it mean that in master (with Cowboy across the board), there is nothing to improve in the implementation?

Dzol commented:
> @Dzol another thing to clarify: this only submits a test. Does it mean that in master (with Cowboy across the board), there is nothing to improve in the implementation?

Briefly: yes, there's nothing to improve.

Looking through the Web Dispatch/Webmachine log would help to find an IP address in cases like the one in the [mailing list conversation](https://groups.google.com/d/msg/rabbitmq-users/CBUYqqBwlg0/nsQwQD8HBQAJ) for https://github.com/rabbitmq/rabbitmq-server/issues/483.

The Management plugin on branch **stable** does things the same way regardless of Cowboy:

```
rabbitmq_management $ git branch
  master
* stable
rabbitmq_management $ grep -r --color webmachine_log:add_handler .
./src/rabbit_mgmt_app.erl:        _    -> webmachine_log:add_handler(webmachine_log_handler, [LogDir])
rabbitmq_management $
```

There should be log like this, though it's in a separate file from the `error_logger` log (which was the one detailed in the [mailing list conversation](https://groups.google.com/d/msg/rabbitmq-users/CBUYqqBwlg0/nsQwQD8HBQAJ)):

```
$ cat $PATH_TO_LOG_DIR/access.log.2016_01_05_07
127.0.0.1 - - [05/Jan/2016:08:31:26 +0100] "GET /wonderland HTTP/1.1" 500 28 "" ""
```

To adapt the test for branch _stable_ we'd need to change only the handler in the test back to the old, not Cowboy, style.

Dzol commented:
> OK, I understand the question better now. Take a look at what test directory is used by rabbitmq-test. Off top of my head, it is $TMPDIR/rabbit-test/log/\* but those differ in multi-node tests.

The most recent push uses `RABBITMQ_LOG_BASE` which is prefixed with `TMPDIR`.

Commit:
Change path to the Web Dispatch log file for the test

Write log file for HTTP requests/responses to the canonical directory
(when testing). This is set when the test set is run.

michaelklishin commented:
Understood, thank you. Will QA it later today.
